pkg/storage/streams/store.go

```diff
@@ -155,24 +232,35 @@ func (s *streamStore) Get(ctx context.Context, path paths.Path) (
 		return nil, Meta{}, err
 	}
 
-	var rangers []ranger.Ranger
+	derivedKey, err := path.DeriveContentKey(s.rootKey)
+	if err != nil {
+		return nil, Meta{}, err
+	}
 
+	var rangers []ranger.Ranger
 	for i := int64(0); i < msi.NumberOfSegments; i++ {
 		currentPath := fmt.Sprintf("s%d", i)
```

Comment:
We should use the new getSegmentPath method here too, right?

Reply:
yep